File: android/app/src/main/java/ae/findmybay/data/api/DeviceApi.kt
package ae.findmybay.data.api

import kotlinx.serialization.Serializable
import retrofit2.http.Body
import retrofit2.http.POST

@Serializable
data class RegisterDeviceBody(
    val fcmToken: String,
    val platform: String = "android",
)

@Serializable
data class LocationBody(
    val lat: Double,
    val lng: Double,
    val accuracyM: Float? = null,
)

@Serializable
data class OkResponse(val ok: Boolean)

interface DeviceApi {
    @POST("devices/register")
    suspend fun register(@Body body: RegisterDeviceBody): OkResponse

    /** Push the customer's last-known location for the smart-alert worker. */
    @POST("devices/location")
    suspend fun reportLocation(@Body body: LocationBody): OkResponse
}

File: ios/FindMyBay/Data/API/DeviceAPI.swift
// DeviceAPI.swift
//
// Device endpoints — parity with `data/api/DeviceApi.kt`.

import Foundation

protocol DeviceAPI {
    func register(fcmToken: String) async throws -> OkResponse
    func reportLocation(lat: Double, lng: Double, accuracyM: Double?) async throws -> OkResponse
}

struct DeviceAPIImpl: DeviceAPI {
    let client: APIClient

    func register(fcmToken: String) async throws -> OkResponse {
        let body = RegisterDeviceBody(fcmToken: fcmToken, platform: "ios")
        return try await client.request(.POST, path: "devices/register", body: body)
    }

    func reportLocation(lat: Double, lng: Double, accuracyM: Double?) async throws -> OkResponse {
        let body = LocationBody(lat: lat, lng: lng, accuracyM: accuracyM)
        return try await client.request(.POST, path: "devices/location", body: body)
    }
}
